packages/polymath-ui/src/components/NoticeBar/NoticeBar.js

```diff
@@ -50,24 +52,50 @@ class NoticeBar extends Component<Props> {
               width="24"
               height="25"
             />
-            {notice.title}
+            <p className="pui-notice-bar-title-text">{notice.title}</p>
           </div>
-          <div
-            className="pui-notice-bar-text"
-            dangerouslySetInnerHTML={{ __html: notice.content }}
-          />
-          <div
-            role="button"
-            className="pui-notice-bar-close"
-            onClick={this.handleClose}
-            tabIndex={0}
-          >
-            <Icon name="close--glyph" fill="#ffffff" width="16" height="16" />
+          <div className="pui-notice-bar-text-container">
+            <p
+              className="pui-notice-bar-text"
+              dangerouslySetInnerHTML={{ __html: notice.content }}
+            />
+          </div>
+          <div className="pui-notice-bar-button-container">
+            <div
+              role="button"
+              className="pui-notice-bar-close"
+              onClick={this.handleClose}
+              tabIndex={0}
+            >
+              <Icon name="close--glyph" fill="#ffffff" width="16" height="16" />
+            </div>
           </div>
         </div>
       </div>
     );
   }
+
+  componentDidUpdate() {
+    /**
+     * NOTE @monitz87: I had to use this fork of the 'clamp-js' library (https://www.npmjs.com/package/clamp-js-main)
+     * because line clamping is horribly unsupported in CSS (and what little support there is is very browser-specific).
+     * What this code does is truncate the title and paragraph (maintaining the integrity of any HTML elements inside it) if they
+     * exceeds 2 lines, adding ellipses to replace the missing text
+     */
+    const paragraph = document.getElementsByClassName('pui-notice-bar-text')[0];
```

Comment:
We should use a Ref here instead of a document selector.

Reply:
Also is there a way we can jut pass this in render? Feels like we are falling out of the React flow

Reply:
The way clamp works means it has to be done post-render AFAIK. But yeah, refs are a good idea